src/events/SDL_events.c

```diff
@@ -906,6 +905,28 @@ SDL_PollEvent(SDL_Event * event)
     return SDL_WaitEventTimeout(event, 0);
 }
 
+void
+SDL_FreeEvent(SDL_Event * event)
+{
+    if (event->type == SDL_DROPFILE)
+    {
+        if (event->drop.file)
+            SDL_free(event->drop.file);
+    }
+    else if (event->type == SDL_TEXTEDITING_EXT)
+    {
+        if (event->editExt.text)
+            SDL_free(event->editExt.text);
+    }
+    else if (SDL_USEREVENT <= event->type && event->type < SDL_userevents)
+    {
+        /* If the event is in the range of registered user events we'll want to execute
+        the custom cleanup function (if one has been provided)*/
+        if (event->user.free && (event->user.data1 || event->user.data2))
+            event->user.free(event->user.data1, event->user.data2);
```

Comment:
```suggestion
            event->user.free(event);
```
I think it's a bit more neat if the entire `event` is passed to `user.free` here. Probably gives more control and context to users. The signature of `user.free` could be the same as `SDL_FreeEvent`